astropy/cosmology/funcs.py

```diff
@@ -14,7 +21,41 @@
 __doctest_requires__ = {'*': ['scipy']}
 
 
-def z_at_value(func, fval, zmin=1e-8, zmax=1000, ztol=1e-8, maxfun=500):
+def _z_at_array(func, fvals, zmin, zmax, nbins=10000,
+                logspace=True, interpolation=None):
+    """Helper function to interpolate (func, z) over a grid for array input"""
+    if logspace:
+        zgrid = np.logspace(np.log10(zmin), np.log10(zmax), nbins)
+    else:
+        zgrid = np.linspace(zmin, zmax, nbins)
+
+    fgrid = func(zgrid)
+
+    fvals_val = fvals.value
+    fgrid_val = fgrid.to_value(fvals.unit)
+
+    if interpolation is None:
+        if USE_SCIPY:
+            interpolation = 'cubic'
+        else:
+            warnings.warn("""\
+SciPy not found, so falling back to linear numpy interpolation scheme
+for array z_at_value, instead of the normal cubic spline.""")
+            interpolation = 'linear'
+
+    assert interpolation in ['cubic', 'linear']
+
+    if interpolation == 'cubic':
```

Comment:
& similarly for "linear".

Reply:
Happy with either - whatever is standard

Reply:
Either is good. Astropy has no formal standard on this that I can find, since sub-packages are maintained by different people. Presumably and APE for the docs would be required to set a package-wide standard. Or something that breaks in the CI to nudge people in a direction (see #11118). I'm perfectly happy *without* the `.lower()`, so long as the allowed keys are noted in the docs. numpydoc recommends literals be in curly braces ``interpolation : {'cubic', 'linear'} or None, optional``

Reply:
Done - I'll note those two options in that numpydoc format, and leave out the `.lower()`.